File: SERVICES/Services1/app/src/main/java/com/javadude/services1/MainActivity.java
package com.javadude.services1;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
	}

	@Override
	protected void onStart() {
		super.onStart();
		Intent intent = new Intent(this, StartedService.class);
		startService(intent);
	}

	@Override
	protected void onStop() {
		Intent intent = new Intent(this, StartedService.class);
		stopService(intent);
		super.onStop();
	}
}

File: Module - Services/Services1/app/src/main/java/com/javadude/services1/MainActivity.kt
package com.javadude.services1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

// Activity that starts a "started" service
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()
        startService(Intent(this, StartedService::class.java))
    }

    override fun onStop() {
        stopService(Intent(this, StartedService::class.java))
        super.onStop()
    }
}

